### PR TITLE
drivers: stepper: tmc: create rampgen Kconfig template

### PR DESCRIPTION
**Pull request by jilaypandya**
create rampstat Kconfig template to enable respective tmc drivers to reuse the common configurations

PR to facilitate #80784

### CODE REVIEW

**Review by fabiobaltieri — APPROVED** [review]

**Review by fabiobaltieri — COMMENTED** [review]
```
doc: migration guide: 4.1:add deprecation info of STEPPER_ADI_TMC_RAMP_GEN
                         ^
                         '--- missing space
```

**Comment by jilaypandya** [comment]
> ```
> doc: migration guide: 4.1:add deprecation info of STEPPER_ADI_TMC_RAMP_GEN
>                          ^
>                          '--- missing space
> ```

Already reached 75 characters :)

File: drivers/stepper/adi_tmc/Kconfig.tmc_rampgen_template
Comment (dipakgmx) [inline drivers/stepper/adi_tmc/Kconfig.tmc_rampgen_template:1]: This deals more with RAMP_STAT and not Ramp Generator.

Comment (jilaypandya) [reply drivers/stepper/adi_tmc/Kconfig.tmc_rampgen_template:1]: Yep I know, however RAMP_STAT is also a Register within Ramp Generator Motion Control Registers, hence I selected this name

**Review by dipakgmx — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
> Already reached 75 characters :)

Is that enforced? I think that's more of a suggestion

**Review by bjarki-andreasen — APPROVED** [review]
cool :)

**Pull request merged**